ctapipe/image/cleaning.py

```diff
@@ -67,16 +73,100 @@ def tailcuts_clean(geom, image, picture_thresh=7, boundary_thresh=5,
     # matrix (2d), we find all pixels that are above the boundary threshold
     # AND have any neighbor that is in the picture
     pixels_above_boundary = image >= boundary_thresh
-    pixels_with_picture_neighbors = geom.neighbor_matrix_sparse.dot(
-        pixels_in_picture)
+    pixels_with_picture_neighbors = geom.neighbor_matrix_sparse.dot(pixels_in_picture)
     if keep_isolated_pixels:
-        return (pixels_above_boundary
-                & pixels_with_picture_neighbors) | pixels_in_picture
+        return (
+            pixels_above_boundary & pixels_with_picture_neighbors
+        ) | pixels_in_picture
     else:
         pixels_with_boundary_neighbors = geom.neighbor_matrix_sparse.dot(
-            pixels_above_boundary)
-        return ((pixels_above_boundary & pixels_with_picture_neighbors) |
-                (pixels_in_picture & pixels_with_boundary_neighbors))
+            pixels_above_boundary
+        )
+        return (pixels_above_boundary & pixels_with_picture_neighbors) | (
+            pixels_in_picture & pixels_with_boundary_neighbors
+        )
+
+
+def mars_cleaning_1st_pass(
+    geom,
+    image,
+    picture_thresh=7,
+    boundary_thresh=[5, 5],
```

Comment:
you probably shouldn't use a mutable type as a default here - it can lead to unexpected behavior since it's initialized at load time, and then can be modified by code later.

Reply:
set it to `None` instead, and in the code check for non and set to `[5,5]` if you really need a default.

Reply:
see for example https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Reply:
I just realized that the version I pushed here is slightly older that that on _protopipe_: there I dropped the list, and instead I fixed both boundary thresholds to the same value which is anyway what MARS is doing.
So boundary_thresh should be actually just 5 (a default value like another).

Reply:
For the new methods I plan to develop to make things more modular, I will make one totally general

Reply:
Done.

Reply:
Use a `tuple`: `(5, 5)` that is immutable and you can still set both thresholds.

Reply:
Thanks for the advice! If it's ok for you, though, since the image cleaning implementation of CTA-MARS is indeed p-b-b there is no need to do this here. I can do it in a more general implementation of a double-boundary-tailcut which will be able to use for any purpose later on.